frontend/packages/kubevirt-plugin/integration-tests/tests/cdi-upload.scenario.ts

```diff
@@ -66,28 +74,102 @@ describe('KubeVirt CDI Upload', () => {
       await uploadForm.upload(pvc);
       await browser.wait(until.presenceOf(errorMessage));
       expect(errorMessage.getText()).toContain('not supported');
-    },
-    CDI_UPLOAD_TIMEOUT_SECS,
-  );
+    });
 
-  it(
-    'ID(CNV-4890) Upload image for golden OS',
-    async () => {
-      const pvc: PVCData = {
-        image: desImage,
-        os: OperatingSystem.RHEL7,
-        pvcSize: '1',
-        pvcSizeUnits: 'Gi',
-        storageClass: STORAGE_CLASS,
-      };
+    it(
+      'ID(CNV-5038) Upload data to golden OS template again after delete',
+      async () => {
+        await rhel7PVC.create();
+        await rhel7PVC.delete();
+        // it can be associated again.
+        await rhel7PVC.create();
+      },
+      2 * CDI_UPLOAD_TIMEOUT_SECS,
+    );
 
-      await uploadForm.upload(pvc);
-      // It has to click the upload button again to trigger uploading based on actual test results.
-      await click(saveChangesBtn);
-      await browser.wait(until.textToBePresentInElement(cdiUploadView.uploadProgress, '100%'));
-      await click(cdiUploadView.viewStatusID);
-      await browser.wait(until.textToBePresentInElement(pvcStatus, PVC_STATUS.BOUND));
-    },
-    CDI_UPLOAD_TIMEOUT_SECS,
-  );
+    it('ID(CNV-5176) It shows an error when uploading data to golden OS again', async () => {
+      await uploadForm.openForm();
+      await uploadForm.selectGoldenOS(rhel7PVC.os);
+      await browser.wait(until.presenceOf(errorMessage));
+      expect(errorMessage.getText()).toContain('Operating system source already defined');
+    });
+  });
+
+  describe('KubeVirt GOLDEN OS Creation', () => {
+    const leakedResources = new Set<string>();
+    const fedoraPVC = new PVC(FEDORA_PVC);
+    const win10PVC = new PVC(WIN10_PVC);
+
+    it(
+      'ID(CNV-5042) Create multiple VMs from the same golden os template',
+      async () => {
+        const vm1 = new VMBuilder(getBasicVMBuilder())
+          .setOS(rhel7PVC.os)
+          .generateNameForPrefix('auto-clone-vm1')
+          .setStartOnCreation(true)
+          .build();
+
+        const vm2 = new VMBuilder(getBasicVMBuilder())
+          .setOS(rhel7PVC.os)
+          .generateNameForPrefix('auto-clone-vm2')
+          .setStartOnCreation(true)
+          .build();
+
+        await rhel7PVC.create();
+        await withResource(leakedResources, vm1.asResource(), async () => {
```

Comment:
It can be done by using nested `withResource`.

Reply:
maybe you could enhance this method or create a 2nd one called `withResources`

Reply:
I met a pitfall by using `withResources` today, the resources cannot work the callback function in this situation.

```
await withResources(leakedResources, [vm1.asResource(), vm2.asResource()], async () => {
          await vm1.create();
          await vm1.navigateToDetail();
          await vm2.create();
          await vm2.navigateToDetail();
        });
```

So I will fall back to nested withResource.

Reply:
I don't understand. What is the issue?

Reply:
If it has multiple resources, the callback will run multiple times for each resource, it does not make sense if the callback is not something simple like `kubect delete`. eg, below steps should not been running for each resource.
```
          await vm1.create();
          await vm1.navigateToDetail();
          await vm2.create();
          await vm2.navigateToDetail();
```

Reply:
why? The callback does not have to run multiple times - just once is enough.

Steps
- add all resources to leakedResources
- run callback

It would be much more readable than nested withResource

Reply:
The first version of withResources I wrote like below, it's not working for multiple resources.
```
export async function withResources(
  resourceSet: Set<string>,
  resources: any,
  callback: Function,
  keepResource: boolean = false,
) {
  resources.forEach(resource => {
    withResource(resourceSet, resource, callback, keepResource);
  });
}
```

To improved version like below, does it look good to you?
```
export async function withResources(
  resourceSet: Set<string>,
  resources: any,
  callback: Function,
  keepResource: boolean = false,
) {
  resources.forEach(resource => 
    addLeakableResource(resourceSet, resource);
  )
  await callback();
  if (!keepResource) {
    resources.forEach(resource => 
      deleteResource(resource);
      removeLeakableResource(resourceSet, resource);
    )
  }
}
```

Reply:
yup!